pkg/minikube/command/exec_runner.go

```diff
@@ -105,12 +114,44 @@ func (*execRunner) Copy(f assets.CopyableFile) error {
 		return errors.Wrapf(err, "error converting permissions %s to integer", f.GetPermissions())
 	}
 
-	return writeFile(dst, f, os.FileMode(perms))
+	switch runtime.GOOS {
+	case "linux":
```

Comment:
please comment what this code is trying to solve

Reply:
This code is trying to solve a problem with bad permissions of /var/lib/minikube/binaries/v1.17.0/* when minikube is run as normal user with the none driver:
When minikube is started as a normal user with the none driver, those files are still created as owned by root and mode 700. And then minikube will fail with "permission denied" (to execute the files there).
The code fixes this issue.

(There might be a better way to fix the issue, but I haven't found one yet.)